clang/test/CXX/drs/cwg26xx.cpp

```diff
@@ -24,6 +40,99 @@ using enum E;
 #endif
 }
 
+namespace cwg2627 { // cwg2627: 19
+#if __cplusplus >= 202002L
+struct C {
+  long long i : 8;
+  friend auto operator<=>(C, C) = default;
+};
+
+void f() {
+  C x{1}, y{2};
+  static_cast<void>(x <=> y);
+  static_cast<void>(x.i <=> y.i);
+}
+
+template<typename T>
+struct CDependent {
+  T i : 8;
+  friend auto operator<=>(CDependent, CDependent) = default;
+};
+
+template<typename T>
+concept three_way_comparable = requires(T t) { { t <=> t }; };
+template<typename T>
+concept bf_three_way_comparable = requires(T t) { { t.i <=> t.i }; };
+static_assert(three_way_comparable<CDependent<long long>>);
+static_assert(bf_three_way_comparable<CDependent<long long>>);
+#endif
+
+#if __cplusplus >= 201103L
+template<typename T, int N>
+struct D {
+  T i : N;
+};
+
+template<typename T, int N>
+D<T, N> d();
+
+#ifdef __SIZEOF_INT128__
+using int64_t = long long;
+static_assert(sizeof(long long) == 8, "long long needs to be 64 bit for this test to work");
+
+int64_t d1{ d<__int128, 63>().i };
+int64_t d2{ d<__int128, 64>().i };
+int64_t d3{ d<__int128, 65>().i };
+// since-cxx11-error@-1 {{non-constant-expression cannot be narrowed from type '__int128' to 'int64_t' (aka 'long long') in initializer list}}
+//   since-cxx11-note@-2 {{insert an explicit cast to silence this issue}}
+#endif
+
+#if __BITINT_MAXWIDTH__ >= 34
+__extension__ _BitInt(33) d4{ d<_BitInt(34), 33>().i };
```

Comment:
`_BitInt` doesn't belong in C++ DR tests. I'd rather see it somewhere in Sema tests for narrowing conversions.